filament/backend/src/opengl/platforms/PlatformEGL.cpp

```diff
@@ -86,13 +86,28 @@ Driver* PlatformEGL::createDriver(void* sharedContext) noexcept {
 
     EGLint major, minor;
     EGLBoolean initialized = eglInitialize(mEGLDisplay, &major, &minor);
+
+    if(!initialized) {
+      EGLDeviceEXT eglDevice;
+      EGLint numDevices;
+
+      PFNEGLQUERYDEVICESEXTPROC eglQueryDevicesEXT =
+              (PFNEGLQUERYDEVICESEXTPROC)eglGetProcAddress("eglQueryDevicesEXT");
+      if (eglQueryDevicesEXT != NULL) {
+          eglQueryDevicesEXT(1, &eglDevice, &numDevices);
+          mEGLDisplay = eglGetPlatformDisplay(EGL_PLATFORM_DEVICE_EXT, eglDevice, 0);
+          initialized = eglInitialize(mEGLDisplay, &major, &minor);
+      }
+    }
+
     if (UTILS_UNLIKELY(!initialized)) {
         slog.e << "eglInitialize failed" << io::endl;
         return nullptr;
     }
 
+#if defined(ANDROID) || defined(FILAMENT_USE_EXTERNAL_GLES3) || defined(__EMSCRIPTEN__)
```

Comment:
it's not right to have these checks on the "generic" PlatformEGL, especially that you have a subclass.